HathorNetwork/hathor-core
# refactor: move simulator utils functions

### Motivation

The events simulator CLI tool was using some functions defined in the `tests` module, which is not copied to the Docker image. To make the tool available via Docker, those functions must be moved to another module.

### Acceptance Criteria

- Move functions from `tests` module to `utils/simulator` module
- Update imports
- Add custom check to make sure `tests` are not imported in `hathor`
- Small changes to fix linter issues with added typings

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged
